### PR TITLE
 refactor: update creator reward recipient logic

### PR DESCRIPTION
**Pull request by kulkarohan**

### CODE REVIEW

**Comment by kulkarohan** [comment]
Current dependencies on/for this PR:
* main
  * **PR #267** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/267" target="_blank">Graphite</a> 
    * **PR #268** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/268" target="_blank">Graphite</a> 
      * **PR #269** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/269" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ourzora/zora-protocol/269?utm_source=stack-comment).

**Comment by oveddan** [comment]
may make sense for this PR to be squash with the one below since theyre kind of the same related work?  you could do that with:
`git checkout update_creator_reward_handling && gt fold`

File: packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol [hunk 1]
```diff
@@ -454,9 +456,18 @@ contract ZoraCreator1155Impl is
...
 
     /// @notice Get the creator reward recipient address
-    /// @dev The creator is not enforced to set a funds recipient address, so in that case the reward would be claimable by creator's contract
-    function getCreatorRewardRecipient() public view returns (address payable) {
-        return config.fundsRecipient != address(0) ? config.fundsRecipient : payable(address(this));
+    /// @param tokenId The token id to get the creator reward recipient for
+    /// @dev The creator is not enforced to set a funds recipient address for a specific token or contract-wide,
+    ///      so in the case of both the reward recipient is set to the creator's contract,
+    ///      which can be withdrawn by the creator via the `withdrawRewards` function.
+    function getCreatorRewardRecipient(uint256 tokenId) public view returns (address payable) {
```
[the remark this answers is not in the document]
Comment (kulkarohan) [reply packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol:463]: yep for frontend

File: packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol [hunk 2]
```diff
@@ -454,9 +456,18 @@ contract ZoraCreator1155Impl is
...
+        if (creatorRewardRecipients[tokenId] != address(0)) {
+            return payable(creatorRewardRecipients[tokenId]);
```
[the remark this answers is not in the document]
Comment (kulkarohan) [reply packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol:465]: done in latest pr - closing this one

File: packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol [hunk 3]
```diff
@@ -686,6 +697,17 @@ contract ZoraCreator1155Impl is
         emit ConfigUpdated(msg.sender, ConfigUpdate.FUNDS_RECIPIENT, config);
     }
 
+    /// @notice Allows the creator reward recipient, token admin, or contract admin to update the address that can claim the token's creator rewards
+    function updateCreatorRewardRecipient(uint256 tokenId, address recipient) external {
+        if (msg.sender != getCreatorRewardRecipient(tokenId) || !_isAdminOrRole(msg.sender, tokenId, PERMISSION_BIT_MINTER)) revert InvalidPermission();
```
[the remark this answers is not in the document]
Comment (kulkarohan) [reply packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol:702]: ended up removing this in the latest pr - closing this one

**Pull request closed without merging**